File: angular-shop/src/app/_models/Order.ts
import {Product} from './Product';

export interface Order {
    id: number;
    customer: number;
    totalOrderValue: number;
    coupon: number;
    products?: Product[];
    orderItems?: OrderItem[];
    cartId?: number;
}


export interface OrderItem {
    id: number;
    order: number;
    product: number;
}

File: play-crud/app/models/Order.scala
package models

import play.api.libs.json.{Json, OFormat}

case class Order(id: Int,
                 customer: Int,
                 totalOrderValue: Float,
                 coupon: Int
                )

object Order {
    implicit val orderFormat: OFormat[Order] = Json.format[Order]
}
